pkg/controller/mongodb/replicaset_controller_test.go

```diff
@@ -301,6 +303,35 @@ func TestAutomationConfig_versionIsNotBumpedWithNoChanges(t *testing.T) {
 	assert.Equal(t, currentAc.Version, 1)
 }
 
+func TestAutomationConfig_CustomMongodConfig(t *testing.T) {
+	mdb := newTestReplicaSet()
+
+	mongodConfig := objx.New(map[string]interface{}{})
+	mongodConfig.Set("net.port", 1000)
+	mongodConfig.Set("arbitrary.config.path", "value")
+	mdb.Spec.AdditionalMongodConfig.Object = mongodConfig
+
+	mgr := client.NewManager(&mdb)
+	r := newReconciler(mgr, mockManifestProvider(mdb.Spec.Version))
+	res, err := r.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{Namespace: mdb.Namespace, Name: mdb.Name}})
+	assertReconciliationSuccessful(t, res, err)
+
+	currentAc, err := getCurrentAutomationConfig(client.NewClient(mgr.GetClient()), mdb)
+	assert.NoError(t, err)
+
+	for _, p := range currentAc.Processes {
+		// Ensure port was overridden
+		assert.Equal(t, float64(1000), p.Args26.Get("net.port").Data())
+
+		// Ensure custom value was added
+		assert.Equal(t, "value", p.Args26.Get("arbitrary.config.path").Data())
+
+		// Ensure default settings went unchanged
+		assert.Equal(t, automationconfig.DefaultMongoDBDataDir, p.Args26.Get("storage.dbPath").Data())
+		assert.Equal(t, mdb.Name, p.Args26.Get("replication.replSetName").Data())
```

Comment:
can we maybe change a net.val1, and ensure that an operator configured net value was untouched? We may be in a situation where the user would need to fully configure what the operator configures if they touch a single field

Reply:
Yes, good idea. I've added another test and it seems as if mergo works nicely without overriding the other.